doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -27,6 +27,29 @@ especially wherever an advanced tactic language is needed. The previous
 implementation of Ltac, described in the previous chapter, will be referred to
 as Ltac1.
 
+Current limitations include:
+
+- There are a number of tactics that are not yet supported in Ltac2 because
+  the interface OCaml and/or Ltac2 notations haven't been written.  See
+  :ref:`defining_tactics`.
+
+- Missing usability features such as:
+
+  - Printing functions are limited and awkward to use.  Only a few data types are
+    printable.
+  - Deep pattern matching and matching on tuples don't work.
+  - If statements on Ltac2 boolean values
+
+- Missing low-level primitives that may be needed for writing automation.  For example,
+  there's no way to get the number of constructors of an inductive type so it's
+  not possible to write automation that's generic over inductive types.  It's not
+  possible to build terms with casts through the low-level API because the cast type
```

Comment:
This is not quite true, you can match on `'I <: True` to extract the cast type.  But it's rather painful.

Reply:
If you can give me a short example I could add that to the chapter as a workaround.

Reply:
I'm not in front of a computer right now, so can't test this, but something like:
```coq
Ltac2 get_vm_cast () :=
  match Constr.Unsafe.kind '(I <: True) with
  | Constr.Unsafe.Cast _ cst _ => cst
  | _ => Control.throw Not_found
  end.
```

Reply:
You need to do something like `let cst := get_vm_cast () in ...`. The message means that you're ignoring the return value of the tactic.

Reply:
This text needs to be updated now that you have the cast example, right?

Reply:
I didn't understand the type case example 100%.  IIUC it shows how to extract info from a cast; it's not a complete example of "A convenient way to **build terms** with casts".  Perhaps that should be described differently.

We don't actually define "cast" in the documentation.  It converts an expression from one type to a compatible more-specific type?  Processed during type checking, not at runtime?

So I don't see how to "get the number of constructors of an inductive type".

After tweaking this, are there other good examples of "Missing low-level primitives that may be needed for writing automation. " or should I drop that bullet point?

Reply:
I added your casting example, though I don't know casts Coq well enough to understand how it works.  In some later PR we'll add a better description of casts.

FWIW, the Ltac2 chapter does say "Unlike Ltac1, there are no type casts at runtime".  That may have been in the back of my mind.

Your other points in this comment are covered now.